### ISSUE TITLE
Container failed to create: feature gate NodeSwap enabled, memorySwap.swapBehavior=UnlimitedSwap

### ISSUE THREAD

**Issue by RichardSufliarsky** [cri-o/cri-o#6786]
### What happened?

I have activated NodeSwap feature gate and have set swapBehaviour to UnlimitedSwap for kubelet. Containers fail to start with the following message:
```
kubelet[1755670]: E0403 12:57:35.635893 1755670 remote_runtime.go:421] "CreateContainer in sandbox from runtime service failed" err="rpc error: code = Unknown desc = container 4ab24f7e3888f4f3ce0e6a8aa2b9875d1e0fc0892aa856faf83f2e86440617b9 create failed because memory swap limit (-1) cannot be lower than memory limit (107374182400)" podSandboxID="39a28ffc1c080925a34bea84469e73996b5b7dd98a4508c6a
a09075d7754bf96"
```
If I change swapBehavior to LimitedSwap, containers will start.

### What did you expect to happen?

Containers should start also with swapBehavior=UnlimitedSwap.

### How can we reproduce it (as minimally and precisely as possible)?

Enable feature gate in /etc/sysconfig/kubelet:
```
KUBELET_EXTRA_ARGS="--feature-gates NodeSwap=true"
```
Changes in /var/lib/kubelet/config.yaml:
```
failSwapOn: false
memorySwap:
  swapBehavior: UnlimitedSwap
```
Restart kubelet:
```
sudo systemctl restart kubelet
```

### Anything else we need to know?

Seems related to this issue https://github.com/cri-o/cri-o/issues/6040 and this PR https://github.com/cri-o/cri-o/pull/6139.

### CRI-O and Kubernetes version

<details>

```console
$ crio --version
crio version 1.24.2
Version:          1.24.2
GitCommit:        bd548b04f78a30e1e9d7c17162714edd50edd6ca
GitTreeState:     clean
BuildDate:        2022-08-09T18:58:47Z
GoVersion:        go1.18.2
Compiler:         gc
Platform:         linux/amd64
Linkmode:         dynamic
BuildTags:        exclude_graphdriver_devicemapper, seccomp
SeccompEnabled:   true
AppArmorEnabled:  false
```

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:22:29Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:15:38Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
```
</details>

### OS version

<details>

```console
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.5 (Ootpa)"
$ uname -a
Linux redis002.lab.cortical.io 4.18.0-348.12.2.el8_5.x86_64 #1 SMP Mon Jan 17 07:06:06 EST 2022 x86_64 x86_64 x86_64 GNU/Linux
```
</details>

### Additional environment details (AWS, VirtualBox, physical, etc.)

<details>
physical
</details>

**Comment by haircommander** [comment]
ah yes the backport needs to be released still, it will be in 1.24.5 https://github.com/cri-o/cri-o/pull/6790

**Comment by hasan4791** [comment]
@RichardSufliarsky We've released 1.24.5 & could you confirm the issue is resolved?
https://github.com/cri-o/cri-o/releases/tag/v1.24.5

**Comment by RichardSufliarsky** [comment]
I confirm that `memorySwap.swapBehavior=UnlimitedSwap` works with cri-o version 1.24.5.
